internal/factory/repository_craft.go

```
@@ -0,0 +1,259 @@
+		return "", ""
+	}
+	return document.Path, document.SHA256
+}
+
+// validatePersistedRepositoryCraft verifies the invocation packet's recorded
+// craft identity before recovery can rebuild a prompt. Legacy packets without
+// repository craft remain compatible and skip this check.
+func validatePersistedRepositoryCraft(run store.Run, invocation store.Invocation) error {
+	packet, err := decodeSpecificationPacket(run.SpecificationPacket)
+	if err != nil {
+		return nil
```

golangci-lint (2.13.2)

[error] error is not nil (line 239) but it returns nil

(nilerr)